src/goose/_logger.py

```diff
@@ -1,9 +1,44 @@
 import logging
 from pathlib import Path
+import json
 
 _LOGGER_NAME = "goose"
 _LOGGER_FILE_NAME = "goose.log"
 
+_TRACE_LOGGER_NAME = "trace_logger"
+_TRACE_LOGGER_FILE_NAME = "trace.log"
+
+TRACE_LEVEL = 5
+logging.addLevelName(TRACE_LEVEL, "TRACE")
+
+
+def trace(self: logging.Logger, message: str, *args: tuple, **kws: dict) -> None:
+    """
+    Log 'message' with severity 'TRACE' to log agent traces.
+
+    Usage: logger.trace("message")
+    """
+    if isinstance(message, dict):
+        try:
+            role = message.get("role", "")
+            content_type = message.get("content", [{}])[0].get("type", "")
+            if content_type == "ToolUse":
+                tool = message.get("content", [{}])[0].get("name", "")
+                content_type = f"{content_type} [{tool}]"
+            content = message.get("content", [{}])[0]
+            formatted_content = "\n" + json.dumps(content, indent=4)
+            formatted_content = formatted_content.replace("\\n", "\n")
+            formatted_message = f"** {role} / {content_type} ** \n{formatted_content}\n\n"
+            message = formatted_message
+        except Exception:
+            message = "\n" + json.dumps(message, indent=4)
+            message = message.replace("\\n", "\n")
+    if self.isEnabledFor(TRACE_LEVEL):
+        self._log(TRACE_LEVEL, message, args, **kws)
+
+
+logging.Logger.trace = trace
```

Comment:
i feel duck typing the `logger` might make it hard to reason later on, especially across different type of handlers. what do you think about extending `Logger` so we can reuse this across other logging handling types?

that way `trace` is available for module-specific logging rather than potentially polluting this specific `trace.log`. it ensures the implementation is also hidden from a this namespace.


```python
class TraceLogger(logging.Logger):
    def __init__(self, name, level=logging.NOTSET):
        super().__init__(name, level)
        logging.addLevelName(TRACE, "TRACE")

    def trace(self, msg, *args, **kwargs):
        if self.isEnabledFor(TRACE):
            self._log(TRACE, msg, args, **kwargs)

logging.setLoggerClass(TraceLogger)
```

Reply:
if we decide `TRACE` is needed, we'll probably also need to add it [here](https://github.com/block-open-source/goose/blob/main/src/goose/cli/main.py#L69)

Reply:
Thanks for these comments - I agree filtering might be a simpler approach. Let me know if the updates are in line with what you were thinking

Reply:
nice! yeah, this makes sense to me :) 

we might want to consider is switching the `FileHandler` to `RotatingFileHandler`/`TimedRotatingFileHandler` so if/when we need to consume these logs it's not too unwieldy

Reply:
I didn't know about these types of filehandlers! Updating to use the TimedRotatingFileHandler